### PR TITLE
refactor: [M3-8244] - Remove `suppressImplicitAnyIndexErrors` and `ignoreDeprecations` Typescript Options

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

Does what it takes to remove `suppressImplicitAnyIndexErrors` and `ignoreDeprecations`  from our `tsconfig.json` 🔧

## Why ❓ 
- See https://github.com/microsoft/TypeScript/issues/51909
- I really want to get this behind us before there are even more code that needs to be refactored
- I'm excited for upcoming features like [Inferred Type Predicates](https://devblogs.microsoft.com/typescript/announcing-typescript-5-5/#inferred-type-predicates) that we currently can't use.
- Some things in the diff might look hacky, but the changes are likely equal to or more type-safe then they were
- Doing this to unblock us from using the latest TypeScript versions 🚧 

> `suppressImplicitAnyIndexErrors`  will continue to "work" until TypeScript 5.5, at which point they will be removed entirely.

## How to test 🧪

- Verify typechecking passes ✅ 
- Scan the changes and look from any changes that could break Cloud Manager
  - Because this PR touches mostly types, there shouldn't be too much customer-facing risk

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Review by jaalah-akamai — COMMENTED** [review]
Barely scratched surface, but I'm going to pull things down and do broader review in a bit.

File: packages/manager/src/components/DownloadCSV/DownloadCSV.tsx [hunk 1]
```diff
@@ -86,7 +86,7 @@ export const cleanCSVData = (data: any): any => {
 
   /** if it's an object, recursively sanitize each key value pair */
   if (typeof data === 'object') {
-    return Object.keys(data).reduce((acc, eachKey) => {
+    return Object.keys(data).reduce<any>((acc, eachKey) => {
```
[the remark this answers is not in the document]
Comment (bnussman-akamai) [reply packages/manager/src/components/DownloadCSV/DownloadCSV.tsx]: Addressed in 9aaa746

**Review by dwiley-akamai — APPROVED** [review]
Type checks passed ✅
No type errors observed locally ✅
No app crashes or broken functionality observed locally ✅

**Review by AzureLatte — APPROVED** [review]
Looks good

**Pull request merged**